pkg/graveler/graveler.go

```diff
@@ -1116,7 +1317,41 @@ func (g *KVGraveler) AddCommit(ctx context.Context, repositoryID RepositoryID, c
 
 // addCommitNoLock lower API used to add commit into a repository. It will verify that the commit meta-range is accessible but will not lock any metadata update.
 func (g *KVGraveler) addCommitNoLock(ctx context.Context, repositoryID RepositoryID, commit Commit) (CommitID, error) {
-	return g.db.addCommitNoLock(ctx, repositoryID, commit)
+	repo, err := g.db.GetRepository(ctx, repositoryID)
+	if err != nil {
+		return "", fmt.Errorf("get repository %s: %w", repositoryID, err)
+	}
+
+	// verify access to meta range
+	ok, err := g.CommittedManager.Exists(ctx, repo.StorageNamespace, commit.MetaRangeID)
+	if err != nil {
+		return "", fmt.Errorf("checking for meta range %s: %w", commit.MetaRangeID, err)
+	}
+	if !ok {
+		return "", fmt.Errorf("%w: %s", ErrMetaRangeNotFound, commit.MetaRangeID)
+	}
+
+	// add commit
+	commitID, err := g.RefManager.AddCommit(ctx, repositoryID, commit)
+	if err != nil {
+		return "", fmt.Errorf("add commit: %w", err)
+	}
+	return commitID, nil
+}
+
+func (g *KVGraveler) stagingEmpty(ctx context.Context, branch *Branch) (bool, error) {
```

Comment:
This is not the correct implementation, and the naming is misleading. The function name should be in the sorts of branchUncommittedExist or branchStagingEmpty and it should return false/true if all tokens (staging + sealed) are empty (Please add a TODO here, or open an issue)

Reply:
added TODO